### ISSUE TITLE
kafka security_protocol Does kafka support SASL_SSL?

### ISSUE THREAD

**Issue by liuwenzhi0318** [apache/apisix#9353]
### Description

![image](https://user-images.githubusercontent.com/37536326/233815120-37921ec7-d2e5-40f8-af2d-d423a5cc8a47.png)

**Comment by liuwenzhi0318** [comment]
![image](https://user-images.githubusercontent.com/37536326/233815148-397928f4-a237-4c54-95a8-1e55dd7ba87d.png)

**Comment by zccabb** [comment]
Yes, Kafka does support SASL_SSL as one of its security protocols.

**Comment by kingluo** [comment]
@liuwenzhi0318 Is your question related to apisix? Do you use apisix as a Kafka proxy and you're wondering does it support SASL_SSL?

**Comment by macdoor** [comment]
I also need this function, I want to use in kafka logger

**Comment by kingluo** [comment]
@liuwenzhi0318 @macdoor 
The Kafka logger or pubsub in apisix supports SASL_SSL, but for the time being, no configuration is provided.

> SASL_SSL simply means that the client authentication (SASL) is used over a protected connection (SSL) to prevent interception instead of over a plain connection.

**Comment by macdoor** [comment]
@liuwenzhi0318 Would you please provide an example using kafka logger with sasl_ssl?

**Comment by kingluo** [comment]
APISIX uses lua-resty-kafka, which has cosocket ssl options for ssl usage. But APISIX did not export them into the plugin configuration, so you can not use SSL now. Maybe we will add them later.

**Comment by macdoor** [comment]
@kingluo This feature is very useful.

**Comment by moonming** [comment]
@macdoor @liuwenzhi0318 Contributions are welcome if you are interested in this feature :)

**Comment by macdoor** [comment]
@moonming I'm interested and have free time, but I have very little development experience with lua+ngx, is this ok?

**Comment by moonming** [comment]
> @moonming I'm interested and have free time, but I have very little development experience with lua+ngx, is this ok?
No worries, @Sn0rt and @kingluo can  help you

**Comment by macdoor** [comment]
@moonming that's great, let's start

**Comment by macdoor** [comment]
@moonming I built a test environment, including apisix 3.4.1 development environment, apisix-dashboard 3.0.1, and kafka 3.5.1, and added a log output code to the existing kafka-logger.lua. The current code can support plaintext and sasl_plaintext modes, but it lacks support for sasl_ssl mode. What should I do next?

**Comment by kingluo** [comment]
> @moonming I built a test environment, including apisix 3.4.1 development environment, apisix-dashboard 3.0.1, and kafka 3.5.1, and added a log output code to the existing kafka-logger.lua. The current code can support plaintext and sasl_plaintext modes, but it lacks support for sasl_ssl mode. What should I do next?

Pass [ssl and ssl_verify](https://github.com/doujiang24/lua-resty-kafka#methods) to lua-resty-kafka.
[As said before](https://github.com/apache/apisix/issues/9353#issuecomment-1590553081), SASL_SSL is just SASL over SSL.

**Comment by macdoor** [comment]
@kingluo I added the ssl and ssl_verify parameters in kafka-logger.lua and passed them to the producer, and now I can send kafka messages through ssl

**Comment by macdoor** [comment]
@liuwenzhi0318 If you want to use apisix to proxy kafka, you can refer to https://juejin.cn/post/7114175646502223903 to configure tls parameters in upstream

**Comment by Revolyssup** [comment]
@macdoor Will you create a PR for it?

**Comment by myselfmayur1234** [comment]
Anyone can let me know, how to supply SSL certificates while using kafka-logger? I updated the kafka-logger code to pass ssl and ssl_verify parameters but getting SSL_read() failed - routines::ssl/tls alert bed certificates.

Configured below directives in config.yaml for client cert, private key and trusted cert:
`lua_ssl_trusted_certificate
 lua_ssl_certificate_key
 lua_ssl_certificate
`
Appreciate your help

**Comment by Baoyuantop** [comment]
Hi @myselfmayur1234, how is your Kafka certificate issued?

**Comment by myselfmayur1234** [comment]
Hi @Baoyuantop Certificates are basically generated using tool "Microsoft AD CS" (Here we used **Venafi**) and then CSR goes for CA signing. Those are valid certificates.

**Comment by Baoyuantop** [comment]
Does your Kafka server need to verify client certificates? lua-resty-kafka doesn't seem to support passing client certificates.

**Comment by myselfmayur1234** [comment]
ok. So you mean for now there is no solution except if we put any secure tunnel in-between e.g. kafka-proxy.

Thanks for your support @Baoyuantop 
Lets wait for updates from APISIX
